src/components/shape/ammo-shape.js

```diff
@@ -0,0 +1,82 @@
+/* global Ammo,THREE */
+const threeToAmmo = require("three-to-ammo");
+
+var AmmoShape = {
+  schema: {
+    type: { default: "hull", oneOf: ["box", "cylinder", "sphere", "capsule", "cone", "hull", "mesh"] },
+    recenter: { default: false }, //recenter the object3D's geometry
+    autoGenerateShape: { default: true }, //disable if using custom halfExtents or sphereRadius
+    mergeGeometry: { default: true }, //use all geometry on the object to generate the shape. Otherwise, setMesh must be called explicitly
```

Comment:
This name is pretty ambiguous to me – seems to imply that the object contains multiple `THREE.Mesh` instances, and that they'll be modified/merged in the process? This property and `autoGenerateShape` seem to gate the exact same code, could they be changed something like `type: none` (preferred) or `manual: true`?

Reply:
This property is really poorly named. @mqp is currently trying to do an optimization pass on my `three-to-ammo` code a bit which may change the need for some of these properties. 

Regardless, the intent is that for some objects, you want to generate a single shape for all the geometry (for each mesh), but in other cases you may want to only generate it the specific mesh that is passed in. This is specifically unique from `autoGenerateShape` which only dictates whether or not it should try to read the geometry of the mesh(es) or use the passed in halfExtents/sphereRadius.

E.g. Hubs will use `mergeGeometry: false` for static environment meshes, so that each geometry on the environment has its own mesh collider which (in theory) will let the physics collisions use occlusion culling when objects are determining if they collide with the environment. One could also use this when generating a convex hull and instead generate a convex hull for each geometry on an object. (In practice this generates a lot of overlapping convex hulls and is less performant that a single convex hull, but I liked keeping the option available.

Reply:
Ok, yeah the need for some kind of option makes sense here... I'm just still struggling with the API naming.

Do you think we could combine the `mergeGeometry` and `autoGenerateShape` options into a single modal property? For example:

```html
<a-entity ammo-shape="type: box; fit: all | compound | manual;" />
```

| fit | description |
|---|---|
| `all` | A single shape is automatically sized to bound all meshes within the entity. |
| `compound` | Multiple shapes are generated to bound each individual mesh within the entity. |
| `manual` | A single shape is sized manually. Requires `halfExtents` or `sphereRadius`. |

I think that covers everything, except the case where you'd use `setMesh`. I'm not 100% sure I understand how you'd use that, but maybe `type: none` would handle it?

Reply:
the intent of setMesh is when you wanted to do what you're calling `compound` (although without necessarily doing it for every submesh if you don't want to). E.g. in Hubs we walk through all the meshes in an environment gltf scene, and create a new `ammo-shape` for each. https://github.com/mozilla/hubs/blob/a621ca51136c40291e5ab557c2511ae2bf971c56/src/utils/media-utils.js#L387
I ideally wanted an `ammo-shape` component to be 1:1 with a single btCollisionShape, so I'm not sure how this would work otherwise. I suppose that's not strictly necessary, and an `ammo-shape` could contain a list of collisionShapes instead of just one. I have to think about this more and reconcile how important it is to be able to exactly specify a mesh vs just assume you'd want to do it for all meshes. I do prefer your suggested API naming over mine though.